### ISSUE TITLE
Incorrect "Expected {} arguments, found {}" error

### ISSUE THREAD

**Issue by SamRodri** [rust-analyzer/rust-analyzer#5380]
This code causes the incorrect error:
```rust
macro_rules! Type {
    () => { u32 };
}
enum Foo {
    Bar(Type![])
}
impl Foo {
    fn new() -> Self {
        Foo::Bar(0) // <- incorrect error here
    }
}
```

This issue maybe similar to #5147

**Comment by kiljacken** [comment]
This is presumably due to us not handling macros in type position correctly yet. This would probably manifest as us expecting zero members in the enum entry instead of the one that is actually there, and reporting that as an error.

**Comment by flodiebold** [comment]
We should probably at least lower that as an error type, instead of just a missing type though.
